File: app/src/main/java/kr/co/yogiyo/rookiephotoapp/diary/db/Converters.java
package kr.co.yogiyo.rookiephotoapp.diary.db;

import android.arch.persistence.room.TypeConverter;

import java.util.Date;

public class Converters {
    @TypeConverter
    public static Date fromTimestamp(Long value) {
        return value == null ? new Date() : new Date(value);
    }

    @TypeConverter
    public static Long dateToTimestamp(Date date) {
        return date == null ? new Date().getTime() : date.getTime();
    }
}

File: app/src/main/java/kr/co/yogiyo/rookiephotoapp/diary/db/Converters.kt
package kr.co.yogiyo.rookiephotoapp.diary.db

import android.arch.persistence.room.TypeConverter

import java.util.Date

object Converters {
    @JvmStatic
    @TypeConverter
    fun fromTimestamp(value: Long?): Date {
        return value?.let { Date(it) } ?: Date()
    }
    @JvmStatic
    @TypeConverter
    fun dateToTimestamp(date: Date?): Long {
        return date?.time ?: Date().time
    }
}
